static/src/javascripts-legacy/projects/common/modules/experiments/ab.js

```diff
@@ -243,9 +246,10 @@ define([
      */
     function registerCompleteEvent(complete) {
         return function initListener(test) {
-            var variantId = segmentUtil.variantIdFor(test);
 
-            if (segmentUtil.isInTest(test)) {
+            var variantId = getTestVariantId(test.id);
```

Comment:
Discussed with @desbo and we've tested that the success event is still successfully fired even if the canRun criteria is false.